### PR TITLE
fix: correct stack in extension-shared console logs; drop console calls in canary/prod builds

### PR DESCRIPTION
**Pull request by alecdwm**
This PR fixes two issues:

1. When calling `log.log()` (imported from `packages/extension-shared/src/log.ts`) the stacktrace in the browser dev tools would always point to the `log.ts` file instead of the code which created the log.  
Now it points to the correct call site.

2. Some calls to `console.log` weren't being caught by `consoleOverride` (from `packages/extension-core/src/util/logging.ts`) in production/canary builds.  
I've replaced it with the `Terser` `drop_console` config parameter, which completely strips the console calls out from the build. Prod logs are now silent 💤

### CODE REVIEW

File: packages/extension-shared/src/log.ts [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
  **/
 export const log = {
-  error: (message: any, ...args: any[]) => DEBUG && console.error(message, ...args),
-  warn: (message: any, ...args: any[]) => DEBUG && console.warn(message, ...args),
-  log: (message: any, ...args: any[]) => DEBUG && console.log(message, ...args),
-  debug: (message: any, ...args: any[]) => DEBUG && console.debug(message, ...args),
+  error: DEBUG && typeof console.error === "function" ? console.error.bind(console) : noop,
+  warn: DEBUG && typeof console.warn === "function" ? console.warn.bind(console) : noop,
+  log: DEBUG && typeof console.log === "function" ? console.log.bind(console) : noop,
+  debug: DEBUG && typeof console.debug === "function" ? console.debug.bind(console) : noop,
```
Comment (alecdwm) [inline packages/extension-shared/src/log.ts:14]: Fix number 1

File: apps/extension/webpack/webpack.prod.js [hunk 2]
```diff
@@ -91,7 +97,17 @@ const config = (env) => {
...
     optimization: {
       minimize: true,
-      minimizer: [new TerserPlugin({ terserOptions: { compress: true } })],
+      minimizer: [
+        new TerserPlugin({
+          terserOptions: {
+            compress: {
+              defaults: true,
+              // Drop any calls to console.error/warn/log/debug from production/canary builds, and when running tests
+              drop_console: dropConsole(env) ? ["error", "warn", "log", "debug"] : false,
```
Comment (alecdwm) [inline apps/extension/webpack/webpack.prod.js:106]: Fix number 2

File: apps/extension/src/common/enableAnyloggerLogsInDevelopment.ts [hunk 3]
```diff
@@ -3,4 +3,4 @@ import "anylogger-loglevel"
 import { DEBUG } from "extension-shared"
 import loglevel from "loglevel"
 
-if (DEBUG) loglevel.setLevel("info")
+loglevel.setLevel(DEBUG ? "info" : "silent")
```
Comment (alecdwm) [inline apps/extension/src/common/enableAnyloggerLogsInDevelopment.ts:6]: Fix number 2

**Review by 0xKheops — APPROVED** [review]
thx

**Pull request merged**